CMakeLists.txt

```diff
@@ -0,0 +1,161 @@
+                "expected for CUDA build, saw ${Torch_VERSION} instead.")
+    endif ()
+elseif (HIP_FOUND)
+    set(VLLM_GPU_LANG "HIP")
+
+    # Importing torch recognizes and sets up some HIP/ROCm configuration but does
+    # not let cmake recognize .hip files. In order to get cmake to understand the
+    # .hip extension automatically, HIP must be enabled explicitly.
+    enable_language(HIP)
+
+    # ROCm 5.X and 6.X
+    if (ROCM_VERSION_DEV_MAJOR GREATER_EQUAL 5 AND
+            NOT Torch_VERSION VERSION_EQUAL ${TORCH_SUPPORTED_VERSION_ROCM})
+        message(WARNING "Pytorch version >= ${TORCH_SUPPORTED_VERSION_ROCM} "
+                "expected for ROCm build, saw ${Torch_VERSION} instead.")
+    endif ()
+else ()
+    message(FATAL_ERROR "Can't find CUDA or HIP installation.")
+endif ()
+
+#
+# Override the GPU architectures detected by cmake/torch and filter them by
+# the supported versions for the current language.
+# The final set of arches is stored in `VLLM_GPU_ARCHES`.
+#
+override_gpu_arches(VLLM_GPU_ARCHES
+        ${VLLM_GPU_LANG}
+        "${${VLLM_GPU_LANG}_SUPPORTED_ARCHS}")
+
+#
+# Query torch for additional GPU compilation flags for the given
+# `VLLM_GPU_LANG`.
+# The final set of arches is stored in `VLLM_GPU_FLAGS`.
+#
+get_torch_gpu_compiler_flags(VLLM_GPU_FLAGS ${VLLM_GPU_LANG})
+
+#
+# Set nvcc parallelism.
+#
+if (NVCC_THREADS AND VLLM_GPU_LANG STREQUAL "CUDA")
+    list(APPEND VLLM_GPU_FLAGS "--threads=${NVCC_THREADS}")
+endif ()
+
+
+#
+# _C extension
+#
+
+# TODO(luka): happy to hardcode, but these are generated so globbing seems fine.
+file(GLOB FLASH_ATTN_GEN_SRCS "csrc/flash_attn/src/flash_fwd_*.cu")
+message(DEBUG "FLASH_ATTN_GEN_SRCS: ${FLASH_ATTN_GEN_SRCS}")
+
+define_gpu_extension_target(
+        _C
+        DESTINATION vllm_flash_attn
+        LANGUAGE ${VLLM_GPU_LANG}
+        SOURCES csrc/flash_attn/flash_api.cpp ${FLASH_ATTN_GEN_SRCS}
+        COMPILE_FLAGS ${VLLM_GPU_FLAGS}
+        ARCHITECTURES ${VLLM_GPU_ARCHES}
+        # pybind does not support SOABI
+        # USE_SABI 3
+        WITH_SOABI
+)
+
+# TODO(luka): remove this once PyBind is removed
+find_library(TORCH_PYTHON_LIBRARY torch_python PATHS "${TORCH_INSTALL_PREFIX}/lib" REQUIRED)
+target_link_libraries(_C PRIVATE ${TORCH_PYTHON_LIBRARY})
+
+target_include_directories(_C PRIVATE
+        csrc/flash_attn
+        csrc/flash_attn/src
+        csrc/cutlass/include)
+
+# remove macros
+target_compile_options(_C PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:-U__CUDA_NO_HALF_OPERATORS__ -U__CUDA_NO_HALF_CONVERSIONS__ -U__CUDA_NO_HALF2_OPERATORS__ -U__CUDA_NO_HALF_CONVERSIONS__>)
+# other options
+target_compile_options(_C PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:--expt-relaxed-constexpr --expt-extended-lambda --use_fast_math>)
+# custom definitions (TODO: both langs?)
+target_compile_definitions(_C PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:FLASHATTENTION_DISABLE_DROPOUT FLASHATTENTION_DISABLE_UNEVEN_K>)
```

Comment:
You should be able to set these with `COMPILE_FLAGS` on `define_gpu_extension_target`